pkg/model/utxo/receipt.go

```diff
@@ -0,0 +1,188 @@
+	opt := iterateOptions(options)
+
+	var innerErr error
+	var i int
+	if err := u.utxoStorage.Iterate([]byte{UTXOStoreKeyPrefixReceipts}, func(key kvstore.Key, value kvstore.Value) bool {
+
+		if (opt.maxResultCount > 0) && (i >= opt.maxResultCount) {
+			return false
+		}
+
+		i++
+
+		rt := &ReceiptTuple{}
+		if err := rt.kvStorableLoad(u, key, value); err != nil {
+			innerErr = err
+			return false
+		}
+
+		return consumer(rt)
+	}); err != nil {
+		return err
+	}
+
+	return innerErr
+}
+
+// ForEachMigratedAtReceiptTuple iterates over all stored receipt tuples for a given migrated at index.
+func (u *Manager) ForEachMigratedAtReceiptTuple(migratedAtIndex uint32, consumer ReceiptTupleConsumer, options ...UTXOIterateOption) error {
+	opt := iterateOptions(options)
+
+	prefix := make([]byte, 5)
+	prefix[0] = UTXOStoreKeyPrefixReceipts
+	binary.LittleEndian.PutUint32(prefix[1:], migratedAtIndex)
+
+	var innerErr error
+	var i int
+	if err := u.utxoStorage.Iterate(prefix, func(key kvstore.Key, value kvstore.Value) bool {
+
+		if (opt.maxResultCount > 0) && (i >= opt.maxResultCount) {
+			return false
+		}
+
+		i++
+
+		rt := &ReceiptTuple{}
+		if err := rt.kvStorableLoad(u, key, value); err != nil {
+			innerErr = err
+			return false
+		}
+
+		return consumer(rt)
+	}); err != nil {
+		return err
+	}
+
+	return innerErr
+}
+
+// ReceiptToOutputs extracts the migrated funds to outputs.
+func ReceiptToOutputs(r *iotago.Receipt, msId *iotago.MilestoneID) ([]*Output, error) {
+	outputs := make([]*Output, len(r.Funds))
+	for outputIndex, migFundsEntry := range r.Funds {
+		entry := migFundsEntry.(*iotago.MigratedFundsEntry)
+		utxoID := OutputIDForMigratedFunds(*msId, uint16(outputIndex))
+		// we use the milestone hash as the "origin message"
+		outputs[outputIndex] = CreateOutput(&utxoID, hornet.MessageIDFromArray(*msId), iotago.OutputSigLockedSingleOutput, entry.Address.(iotago.Address), entry.Deposit)
+	}
+	return outputs, nil
+}
+
+// OutputIDForMigratedFunds returns the UTXO ID for a migrated funds entry given the milestone containing the receipt
+// and the index of the entry.
+func OutputIDForMigratedFunds(milestoneHash iotago.MilestoneID, outputIndex uint16) iotago.UTXOInputID {
+	var utxoID iotago.UTXOInputID
+	copy(utxoID[:], milestoneHash[:])
+	binary.LittleEndian.PutUint16(utxoID[len(utxoID)-2:], outputIndex)
+	return utxoID
+}
+
+// TreasuryTransactionOutputAmount returns the amount of the treasury output within the treasury transaction.
+func TreasuryTransactionOutputAmount(tx *iotago.TreasuryTransaction) uint64 {
+	return tx.Output.(*iotago.TreasuryOutput).Amount
+}
+
+// ReceiptToTreasuryMutation converts a receipt to a treasury mutation tuple.
+func ReceiptToTreasuryMutation(r *iotago.Receipt, unspentTreasuryOutput *TreasuryOutput, newMsId *iotago.MilestoneID) (*TreasuryMutationTuple, error) {
+	newOutput := &TreasuryOutput{
+		Amount: r.Transaction.(*iotago.TreasuryTransaction).Output.(*iotago.TreasuryOutput).Amount,
```

Comment:
why not use `TreasuryTransactionOutputAmount` instead? Else that func is unused, so remove it.